### ISSUE TITLE
mock.module() does not work on default exports

### ISSUE THREAD

**Issue by JeffreyATW** [oven-sh/bun#6884]
### What version of Bun is running?

1.0.8

### What platform is your computer?

Darwin 23.0.0 x86_64 i386

### What steps can reproduce the bug?

file.ts:
```
export const named = "named";
export default "default";
```

file.test.ts:
```
import { expect, mock, test } from "bun:test";
import file, { named } from "./file";

test("should mock named and default exports", () => {
  expect(named).toBe("named");
  mock.module("./file.ts", () => ({
    default: "mocked default",
    named: "mocked named",
  }));
  expect(named).toBe("mocked named"); // passes
  expect(file).toBe("mocked default"); // fails, value is "default"
});
```

### What is the expected behavior?

Tests pass

### What do you see instead?

Tests fail

### Additional information

This happens when only attempting to mock default, as well. Re-importing the file makes no difference.

**Comment by Jarred-Sumner** [comment]
Context for anyone wanting to work on this

I suspect we need to use `vm.commonIdentifiers.defaultKeyword` somewhere when calling overrideExportValue

**Comment by Jarred-Sumner** [comment]
Fixed by #7129 

With the caveat that default exports which previously did not exist will continue to not exist if the module has already been created. Need to find a workaround in JSC to support that case.
